agent/utils/utils.go

```diff
@@ -208,3 +210,20 @@ func IsAWSErrorCodeEqual(err error, code string) bool {
 	awsErr, ok := err.(awserr.Error)
 	return ok && awsErr.Code() == code
 }
+
+// MapToTags converts a map to a slice of tags.
+func MapToTags(tagsMap map[string]string) []*ecs.Tag {
```

Comment:
FFTI: I'd consider moving this and the other tagging-specific functions (mergeTags) into a tagging-specific file / package.